packages/next/src/lib/metadata/generate/meta.tsx

```diff
@@ -32,6 +32,24 @@ type MultiMetaContent =
   | null
   | undefined
 
+function camelToSnake(camelCaseStr: string) {
+  return camelCaseStr.replace(/([A-Z])/g, function (match) {
+    return '_' + match.toLowerCase()
+  })
+}
+
+function getMetaKey(prefix: string, key: string) {
+  // Use `twitter:image` and `og:image` instead of `twitter:image:url` and `og:image:url`
+  // to be more compatible as it's a more common format
+  if ((prefix === 'og:image' || prefix === 'twitter:image') && key === 'url') {
+    return prefix
+  }
+  if (prefix.startsWith('og:') || prefix.startsWith('twitter:')) {
+    key = camelToSnake(key)
```

Comment:
If there are only limited possibilities maybe we can inline them all?

Reply:
sounds good, I can patch this later in other PRs